packages/dds/tree/src/feature-libraries/chunked-forest/codec/chunkDecoding.ts

```diff
@@ -217,13 +227,56 @@ export class NestedArrayDecoder implements ChunkDecoder {
  */
 export class InlineArrayDecoder implements ChunkDecoder {
 	public constructor(private readonly shape: EncodedInlineArray) {}
-	public decode(decoders: readonly ChunkDecoder[], stream: StreamCursor): TreeChunk {
+	public decode(
+		decoders: readonly ChunkDecoder[],
+		stream: StreamCursor,
+		getIncrementalFieldBatch?: (fieldKey: string) => EncodedFieldBatch,
+	): TreeChunk {
 		const length = this.shape.length;
 		const decoder = decoders[this.shape.shape] ?? oob();
 		const chunks: TreeChunk[] = [];
 		for (let index = 0; index < length; index++) {
-			chunks.push(decoder.decode(decoders, stream));
+			chunks.push(decoder.decode(decoders, stream, getIncrementalFieldBatch));
+		}
+		return aggregateChunks(chunks);
+	}
+}
+
+/**
+ * Decoder for {@link EncodedIncrementalShape}s.
+ */
+export class IncrementalFieldDecoder implements ChunkDecoder {
+	public constructor(
+		private readonly shape: EncodedIncrementalShape,
+		private readonly cache: DecoderContext<EncodedChunkShape>,
+	) {}
+	public decode(
+		_: readonly ChunkDecoder[],
+		stream: StreamCursor,
+		getIncrementalFieldBatch?: (fieldKey: string) => EncodedFieldBatch,
+	): TreeChunk {
+		const data = readStream(stream);
+		if (typeof data === "number") {
+			assert(data === 0, "expected 0 for empty incremental blob");
+			return emptyChunk;
+		}
+
+		assert(typeof data === "string", "expected incremental blob id");
+		const batch = getIncrementalFieldBatch?.(data);
+		assert(batch !== undefined, `Can't find incremental field batch for field ${data}`);
```

Comment:
This is an invalid use of our assert utility. For assert tagging to work, you must have a constant message and no use string interpolation. https://github.com/microsoft/FluidFramework/pull/24013 would provide a way around that for development builds, but that doesn't exist yet.